lib/charms/vault_k8s/v0/vault_kv.py

```diff
@@ -447,47 +400,40 @@ def remove_unit_credentials(self, relation: ops.Relation, nonce: str | Iterable[
         for n in nonce:
             credentials.pop(n, None)
 
-        relation.data[self.charm.app]["credentials"] = json.dumps(credentials, sort_keys=True)
+        self.juju_facade.set_app_relation_data(
+            relation_name=self.relation_name,
+            relation=relation,
+            data={"credentials": json.dumps(credentials, sort_keys=True)},
+        )
 
     def get_credentials(self, relation: ops.Relation) -> dict:
```

Comment:
This may not be quite relevant to this PR but it feels like get_credentials should return a dict, the structure of this dict will always be the same

Reply:
I'm not sure I get this comment, the function returns a dict.

Reply:
Oops, I meant an object (instead of a dict) - edited